greenpithumb/pump_scheduler.py

```diff
@@ -0,0 +1,31 @@
+import clock
+
+
+class PumpScheduler(object):
+    """A wrapper for managing pump scheduling."""
+
+    def __init__(self, clock, sleep_windows):
+        """Creates new PumpScheduler wrapper.
+        
+        Args:
+            clock: A clock interface
+            sleep_windows: A tuple of tuples, each consisting of a sleep 
+                window. Time zone should match that of the clock interface.
+        """
+        self._clock = clock
+        self._sleep_windows = sleep_windows
+
+    def is_running_pump_allowed(self):
+        """Returns True if OK to run pump, otherwise False."""
+        time = self._clock.now()
+        hour = time.hour
+
+        for sleep_hour, wake_hour in self._sleep_windows:
+            if wake_hour < sleep_hour:
+                if hour >= sleep_hour or hour < wake_hour:
```

Comment:
I don't think we can do this, because sleep_hour > hour if hour is midnight or after.